.github/workflows/latest_import.yml

```diff
@@ -0,0 +1,129 @@
+        # set `linter.minImport option` to true in `lakefile`
+        sed -i '/⟨'\`'linter.style.missingEnd, true⟩,/{s=^=  ⟨`linter.minImports, true⟩,\n=}' lakefile.lean
+        # import the `minImport` linter in `Mathlib.Init`
+        sed -i -z 's=^=import Mathlib.Tactic.Linter.MinImports\n=' Mathlib/Init.lean
+        # remove the `Mathlib.Init` import from the `minImport` command to avoid a loop
+        sed -i '/import Mathlib.Init/d' Mathlib/Tactic/MinImports.lean
+        # stage the changes in git so that `git diff` can confirm what changed
+        git add -u
+        git diff HEAD #lakefile.lean Mathlib/Init.lean Mathlib/Init.lean Mathlib/Tactic/MinImports.lean
+
+    - name: build mathlib
+      id: build
+      uses: liskin/gh-problem-matcher-wrap@v3
+      with:
+        linters: gcc
+        run: |
+          lake build
+
+    - name: Full report
+      run: |
+        lake build | sed -z 's=\n\n*\([^⚠w]\)= \1=g' |
+          awk -F: 'BEGIN{max=0}
+            /^⚠/{ print currMax; max=0 }
+            (max < $3) {
+              gsub(/ *Now redun.*/, "")
+              gsub(/ to \[[^]]*\]/, "")
+              gsub(/ *note: this linter.*/, "")
+              gsub(/\.\//, "")
+              gsub(/ *Imports increased by */, "")
+              gsub(/ *New imports */, "")
+              currMax=$0
+              max=$3+0
+          }' |
+          sort --field-separator=: -k3 -nr |
+          awk -F: 'BEGIN{printf("|File|Line|Import increase|New imports|\n|:-|-:|-:|:-|\n")} {
+            printf("| %s | %s | %s | %s |\n", $2, $3, $5, $6)
+          }'
+
+    - name: Zulip report
+      id: late_importers
+      run: |
+        printf $'summary<<EOF\n%s\nEOF' "$(
+          lake build | sed -z 's=\n\n*\([^⚠w]\)= \1=g' |
+            awk -F: 'BEGIN{max=0}
+              /^⚠/{ print currMax; max=0 }
+              (max < $3) {
+                gsub(/ *Now redun.*/, "")
+                gsub(/ to \[[^]]*\]/, "")
+                gsub(/ *note: this linter.*/, "")
+                gsub(/\.\//, "")
+                gsub(/ *Imports increased by */, "")
+                gsub(/ *New imports */, "")
+                currMax=$0
+                max=$3+0
+            }' |
+            sort --field-separator=: -k3 -nr |
+            awk -F: 'BEGIN{
+              printf("|File|Line|Import increase|New imports|\n|:-|-:|-:|:-|\n")
+              con=1
+            }
+              ((con <= 15) && (10 <= $5+0)) {
+                con++
+                printf("| %s | %s | %s | %s |\n", $2, $3, $5, $6)
+            }'
+          )"
+
+    - name: Post output to Zulip
+      uses: zulip/github-actions-zulip/send-message@v1
+      with:
+        api-key: ${{ secrets.ZULIP_API_KEY }}
+        email: 'github-mathlib4-bot@leanprover.zulipchat.com'
+        organization-url: 'https://leanprover.zulipchat.com'
+        to: 'mathlib4'
+        type: 'stream'
+        topic: Late importers report
+        content: ${{ steps.late_importers.outputs.summary }}
```

Comment:
Could you add a cleanup step to delete any files created by this build so that they don't accumulate on the Hoskinson runners?

Reply:
Actually, I guess this isn't necessary because the first cleanup step already nukes everything?